kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/FormServiceBase.java

```diff
@@ -0,0 +1,139 @@
+        this.definitionService = definitionService;
+        this.deploymentService = deploymentService;
+        this.dataService = dataService;
+        this.userTaskService = userTaskService;
+        this.marshallerHelper = new MarshallerHelper(context);
+    }
+
+    public void setProviders(Set<FormProvider> providers) {
+        this.providers = providers;
+    }
+
+    public String getFormDisplayProcess(String containerId, String processId, String lang) {
+        ProcessDefinition processDesc = definitionService.getProcessDefinition(containerId, processId);
+        if (processDesc == null) {
+            throw new IllegalStateException("Process definition " + containerId + " : " + processId + " not found");
+        }
+        Map<String, String> processData = definitionService.getProcessVariables(containerId, processId);
+
+        if (processData == null) {
+            processData = new HashMap<String, String>();
+        }
+
+        Map<String, Object> renderContext = new HashMap<String, Object>();
+        renderContext.put("process", processDesc);
+        renderContext.put("outputs", processData);
+        renderContext.put("marshallerContext", marshallerHelper);
+        renderContext.put("lang", lang);
+
+        for (FormProvider provider : providers) {
+            String template = provider.render(processDesc.getName(), processDesc, renderContext);
+            if (!StringUtils.isEmpty(template)) {
+                return template;
+            }
+        }
+
+        logger.warn("Unable to find form to render for process '{}'", processDesc.getName());
+        return "";
+    }
+
+    public String getFormDisplayTask(long taskId, String lang) {
+        Task task = userTaskService.getTask(taskId);
+        if (task == null) {
+            return "";
+        }
+        String name = task.getName();
+        ProcessDefinition processDesc = dataService.getProcessesByDeploymentIdProcessId(task.getTaskData()
+                .getDeploymentId(), task.getTaskData().getProcessId());
+        Map<String, Object> renderContext = new HashMap<String, Object>();
+
+        Map<String, Object> input = userTaskService.getTaskInputContentByTaskId(taskId);
+        renderContext.put("inputs", input);
+        for (Map.Entry<String, Object> inputVar : ((Map<String, Object>) input).entrySet()) {
+            renderContext.put(inputVar.getKey(), inputVar.getValue());
+        }
+        renderContext.put("lang", lang);
+
+        for (FormProvider provider : providers) {
+            String template = provider.render(name, task, processDesc, renderContext);
+            if (!StringUtils.isEmpty(template)) {
+                return template;
+            }
+        }
+
+        logger.warn("Unable to find form to render for task '{}' on process '{}'", name, processDesc.getName());
+        return "";
+    }
+
+    protected ContentMarshallerContext getMarshallerContext(String deploymentId, String processId) {
```

Comment:
move this method to jbpm (as part of the provider impl) as it's internal part of it and while it is unlikely to change it still can :)